crates/hir_ty/src/lower.rs

```diff
@@ -1259,3 +1260,10 @@ pub(crate) fn return_type_impl_traits(
         Some(Arc::new(Binders::new(num_binders, return_type_impl_traits)))
     }
 }
+
+pub(crate) fn lower_to_chalk_mutability(m: hir_def::type_ref::Mutability) -> Mutability {
+    match m {
+        hir_def::type_ref::Mutability::Shared => Mutability::Not,
+        hir_def::type_ref::Mutability::Mut => Mutability::Mut,
+    }
+}
```

Comment:
This makes me wonder if it would be alright to add `chalk_ir` as a dependency to `hir_def`, then we could implement conversion methods on the `hir_def` types directly and wouldn't need this free functions which I assume will get a few more friends(I think it already has for scalars)

Reply:
Or rather in this case `hir_def`s Mutability could be entirely replaced in that case then, I'm not sure if I'm in favor of that.

Reply:
I think these kinds of functions are what the `lower` module is all about, so this seems fine to me.